staging/src/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go

```diff
@@ -75,6 +76,14 @@ var (
 		},
 		[]string{"resource"},
 	)
+	resourceSizeEstimate = compbasemetrics.NewGaugeVec(
```

Comment:
Out of curiosity, do you think using a ConstHistogram would be possible here? I am wondering if somehow we could get percentiles from this metric.

For example having something like that:
```go
    histogram, err := prometheus.NewConstHistogram(
        prometheus.NewDesc(
            "const_histogram",
            "const histogram metric",
            []string{"group", "resource"},
            nil,
        ),
        100, // count
        1234.56, // sum
        map[float64]uint64{ // buckets
            0.1:  10,
            0.5:  30,
            1.0:  60,
            5.0:  90,
            10.0: 100,
        },
    )
```
and then update the count, sum and bucket values based on put/delete

Reply:
We could, however it would take some work to make it efficient. The current average values was introduced as APF needs it. For this PR I just wanted to expose it, don't know if I will have time to implemented a full histogram.